### DISCUSSION TITLE
issue: after unmount of input it takes another input value

### DISCUSSION THREAD

**Discussion by Mighty683** [react-hook-form/react-hook-form#8137] [Q&A] [answered]
### Version Number

7.29.0

### Codesandbox/Expo snack

https://codesandbox.io/s/loving-bell-kvewjw

### Steps to reproduce

1. Fill first input.
2. Go to second step.
3. Fill second input with different value.
4. Return to first step.
5. Now first input display second input value but values from `watch` are different.

### Expected behaviour

After returning to first step values displayed in input match with state of form.

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
- there is no watch API usage
![Screen Shot 2022-04-05 at 8 30 28 pm](https://user-images.githubusercontent.com/10513364/161735081-00e9a9c3-d4c6-46b5-9ceb-20961b64c09b.png)

- you should composite your component better, and move the nested component to its own (as it will get mounted and remount with each render `renderStep`)

**Comment by Mighty683** [comment]
> * there is no watch API usage
>   ![Screen Shot 2022-04-05 at 8 30 28 pm](https://user-images.githubusercontent.com/10513364/161735081-00e9a9c3-d4c6-46b5-9ceb-20961b64c09b.png)
> * you should composite your component better, and move the nested component to its own (as it will get mounted and remount with each render `renderStep`)

Sorry I pasted wrong URL, here is correct one:

https://codesandbox.io/s/loving-bell-kvewjw

I created this component ad hoc for issue reproduction.

**Reply by bluebill1049** [reply]
give a try on the comment i left below and let me know if that resolves your question.

**Comment by bluebill1049** [comment, marked as answer]
> you should composite your component better, and move the nested component to its own (as it will get mounted and remount with each render renderStep)

this is most likely the root cause, refer to my comment above.

![Screen Shot 2022-04-05 at 8 37 12 pm](https://user-images.githubusercontent.com/10513364/161736182-1334dc6f-6f58-43b1-8d91-185509f21806.png)

**Reply by bluebill1049** [reply]
renderSteop's inner components will get mounted and unmount on each render, move it out and make it a separate component.

**Reply by Mighty683** [reply]
Thank you it's solving problem. I though that I will shortcut with render function because in my case form was really small.

**Reply by bluebill1049** [reply]
no worries, I only have components within the component when I am 100% sure unmount and the mount will not affect the result and without any performance penalty. it's a good thing to be mindful.  👍
